datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -126,19 +138,85 @@ private String isNodeRequestValid(StackV4Response stack, DatalakeHorizontalScale
             return "Negative nodeCount is not accepted. If you want to downsacle your cluster please use lower number than the acutal node count";
         }
         String target = request.getGroup();
-        boolean hostGroupNotExists = stack.getInstanceGroups().stream()
-                .map(InstanceGroupV4Response::getName)
-                .noneMatch(target::equalsIgnoreCase);
-        if (hostGroupNotExists) {
+        if (!stack.getStatus().isAvailable()) {
+            return String.format("Stack is not healthy. Please make sure the stack is healthy before run the Datalake horizontal scaling. " +
```

Comment:
```
        if (!isClusterhealthy(stackResponse, request)) {
            throw new BadRequestException("The cluster is unhealthy");
        }
```
vs.
```
        if (!stack.getStatus().isAvailable()) {
            return String.format("Stack is not healthy. Please make sure the stack is healthy before run the Datalake horizontal scaling. " +
```

I believe stack and cluster status are now unified and it's enough to check the cluster's status in such cases.